BitwardenShared/Core/Platform/Services/ClientService.swift

```diff
@@ -97,8 +100,11 @@ extension ClientService {
 
     /// Returns a `ClientGeneratorsProtocol` for generator data tasks.
     ///
-    func generators() async throws -> ClientGeneratorsProtocol {
-        try await generators(for: nil)
+    /// - Parameter isPreAuth: Whether the client is being used for a user prior to authentication
+    ///     (when the user's ID doesn't yet exist). This primarily will happen in SSO flows.
+    ///
+    func generators(isPreAuth: Bool) async throws -> ClientGeneratorsProtocol {
```

Comment:
🎨 I think we could default this to `false` here if we don't want to always specify it when not in pre-auth flows. That would match the `auth()` function, although I think we could also default that rather than specify an alternate function.

https://github.com/bitwarden/ios/blob/e8fc3661429e368d25fa06db43eba195d4be7a30/BitwardenShared/Core/Platform/Services/ClientService.swift#L74-L86

Reply:
🤔 I see pros and cons with any of the approaches because if we provide default values then someone could introduce a bug by copy/pasting a `generators()` call without parameter and not knowing they need to specify `isPreAuth: false` for that on pre auth flows. OTOH, it feels great not to specify the parameter as it would be `false` in the majority of use cases and it introduces less code changes.

🤔 Also I've been thinking on this for a while on trying not to pass the parameter at all and have some "context" where we can check whether the current flow is actually pre-auth or not so we reduce the chance of introducing bugs on these calls. I have some ideas but need to keep thinking on some cases, for now I see any of the two approaches working fine and we'd just need to keep an eye when calling it and on the PR reviews to be sure we're in the right flow.

Reply:
@matt-livefront 
I missed that, and I agree—though I don't think you can provide defaults in a protocol. It makes sense for both `auth` and `generator` to match here, though.

@fedemkr 
I would definitely much rather some sort of broader context that we can check, rather than passing this around so much. I originally wrote it to just add that flag to *all* of the calls and it was a massive spiderweb of them.

Some of this is also motivated by the fact that in Debug modes, if we log an error we intentionally crash; the issue this is fixing wouldn't actually ever happen in a Release.

Reply:
@KatherineInCode This function is in the extension, which I think would allow adding a default value.

@fedemkr I think the context idea is a good one. In regard to default values, between crashlytics logging and crashing in debug builds these pretty easy to find if you miss a spot. And we don't have too many spots where we use the SDK prior to authentication, so I like the default value for those reasons.

Reply:
@matt-livefront Oh! I see what you mean! Yes, I like that idea and can do something to that effect